### PR TITLE
Run platform instead of parallel tests on IBM JDK

### PR DESCRIPTION
**Pull request by blindpirate**
## Context

We observed frequent internal errors and timeouts on IBM JDK recently:

- https://github.com/gradle/gradle-private/issues/1493
- https://github.com/gradle/gradle-private/issues/1491

They might be related - some unexpected error in test worker may cause worker process hangs. According to the error message, it's very similar to http://www-01.ibm.com/support/docview.wss?uid=swg1IJ00181. However, the bug report alleges this issue has been fixed by `8.0.5.0` but we still see this error on `8.0.5.20`:

```
root@dev91 /opt/files/jdk-linux/ibm-java-sdk-8.0-5.20-x86_64-archive.bin # bin/java -version
java version "1.8.0_181"
Java(TM) SE Runtime Environment (build 8.0.5.20 - pxa6480sr5fp20-20180802_01(SR5 FP20))
IBM J9 VM (build 2.9, JRE 1.8.0 Linux amd64-64-Bit Compressed References 20180731_393394 (JIT enabled, AOT enabled)
OpenJ9   - bd23af8
OMR      - ca1411c
IBM      - 98805ca)
JCL - 20180719_01 based on Oracle jdk8u181-b12
```

Let's first make IBM JDK tests non-parallel and see if everything goes well. If not, we probably need to drop the test coverage for IBM JDK.

To test parallel Gradle execution, this PR changes another configuration to parallel, otherwise there will be no parallel tests at all.

### CODE REVIEW

File: .teamcity/Gradle_Check/model/CIBuildModel.kt [hunk 1]
```diff
@@ -44,8 +44,8 @@ data class CIBuildModel (
                     functionalTests = listOf(
                             TestCoverage(TestType.quickFeedbackCrossVersion, OS.linux, JvmVersion.java8),
                             TestCoverage(TestType.quickFeedbackCrossVersion, OS.windows, JvmVersion.java8),
-                            TestCoverage(TestType.platform, OS.linux, JvmVersion.java10),
-                            TestCoverage(TestType.parallel, OS.linux, JvmVersion.java8, JvmVendor.ibm))
+                            TestCoverage(TestType.parallel, OS.linux, JvmVersion.java10),
+                            TestCoverage(TestType.platform, OS.linux, JvmVersion.java8, JvmVendor.ibm))
```
Comment (wolfs) [inline .teamcity/Gradle_Check/model/CIBuildModel.kt]: Shall we still try to fix the problems with parallel test execution and IBM JDK 8?

@oehme Shall we drop test coverage for IBM JDK altogether? Can we look at our statistics how many people are using IBM JDK to run Gradle?

Comment (blindpirate) [reply .teamcity/Gradle_Check/model/CIBuildModel.kt]: I really doubt if we can fix it. According to the error:

```
Caused by: java.lang.NullPointerException
	at java.io.File.<init>(File.java:288)
	at java.lang.Package$1.run(Package.java:578)
	at java.lang.Package$1.run(Package.java:571)
	at java.security.AccessController.doPrivileged(AccessController.java:640)
	at java.lang.Package.defineSystemPackage(Package.java:571)
	at java.lang.Package.getSystemPackages(Package.java:562)
	at java.lang.VMAccess.getSystemPackages(VMAccess.java:154)
	at com.ibm.oti.vm.BootstrapClassLoader.getPackages(BootstrapClassLoader.java:100)
	at java.lang.ClassLoader.getPackages(ClassLoader.java:1205)
	at java.lang.ClassLoader.getPackages(ClassLoader.java:1208)
	at worker.org.gradle.internal.classloader.FilteringClassLoader$RetrieveSystemPackagesClassLoader.getPackages(FilteringClassLoader.java:63)
```

This is out of our control. I suspect under certain conditions [this issue](http://www-01.ibm.com/support/docview.wss?uid=swg1IJ00181) is triggered again. I'm fine with not testing IBM JDK completely.

Comment (oehme) [reply .teamcity/Gradle_Check/model/CIBuildModel.kt]: @eriwen would have the statistics.

IBM JDK is now called OpenJ9 and hosted at the Eclipse foundation. Maybe we should test that in a more recent version instead.

Comment (blindpirate) [reply .teamcity/Gradle_Check/model/CIBuildModel.kt]: I just realized we're not using latest IBM JDK - latest is `8.0.5.21` but we're using `8.0.5.20`. I just upgraded, let's keep this pr open and see if situation gets improved.

Comment (blindpirate) [reply .teamcity/Gradle_Check/model/CIBuildModel.kt]: It doesn't seem to be improved at all after the upgrading: https://github.com/gradle/gradle-private/issues/1491

I don't know if this is caused by parallel execution, I'd say let's first make it non-parallel, then remove test coverage for IBM JDK totally if it still occurs.

Comment (oehme) [reply .teamcity/Gradle_Check/model/CIBuildModel.kt]: Have you considered what I proposed above?

Comment (blindpirate) [reply .teamcity/Gradle_Check/model/CIBuildModel.kt]: @oehme You mean "more recent version"? The version we're currently testing is released on Sep 3, 2018: https://developer.ibm.com/javasdk/downloads/sdk8/

Are you saying we try OpenJ9 latest released on Aug 14, 2018 https://adoptopenjdk.net/releases.html?variant=openjdk8&jvmVariant=openj9 ?

**Comment by blindpirate** [comment]
It happens again: https://builds.gradle.org/viewLog.html?buildId=15846717&buildTypeId=Gradle_Check_Parallel_Java8_Ibm_Linux_testingJvm

Let's remove IBM JDK for now and add some alternatives later.

**Pull request merged**